src/main/java/com/mraof/minestuck/client/sounds/MusicPlayerOnPlayerSoundInstance.java

```diff
@@ -0,0 +1,37 @@
+package com.mraof.minestuck.client.sounds;
+
+import com.mraof.minestuck.inventory.musicplayer.IMusicPlaying;
+import net.minecraft.client.resources.sounds.AbstractTickableSoundInstance;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.player.Player;
+
+
+public class MusicPlayerOnPlayerSoundInstance extends AbstractTickableSoundInstance
+{
+	private final Player player;
+	private final IMusicPlaying soundEvent;
```

Comment:
You should not be holding on to an `IMusicPlaying` instance like this. You should hold on to the lazy optional of it instead.

Reply:
How can I access get`CurrentMusic()` from something like `LazyOptional<IMusicPlaying> soundEvent` then?

Reply:
The simplest way of doing it is calling `lazyOptional.getValue()`, and it'll return the `IMusicPlaying` instance if it is present, or null if it's not present (and I assume that the sound instance should be stopped when the capability instance is no longer present)